### ISSUE TITLE
Anonymous enums declared in separate types can have conflicting enum member symbols

### ISSUE THREAD

**Issue by ashn-dot-dev** [ashn-dot-dev/sunder#139]
```
~/test$ cat test.sunder
struct foo {
    var kind: enum {
        A;
        B;
    };
}

struct bar {
    var kind: enum {
        B;
        C;
    };
}

func main() void {
}
~/test$ sunder-compile test.sunder 
[test.sunder:10] error: redeclaration of `B` previously declared at [test.sunder:4]
        B;
        ^
```

This does not occur with enums declared as named types:

```
~/test$ cat test.sunder
enum foo {
    A;
    B;
}

enum bar {
    B;
    C;
}

func main() void {
}
~/test$ sunder-compile test.sunder 
~/test$ echo $?
0
```

**Comment by ashn-dot-dev** [comment]
Same thing occurs for top-level anonymous enums.

```
~/test$ cat test.sunder
var foo: enum {
    A;
    B;
} = uninit;

var bar: enum {
    B;
    C;
} = uninit;

func main() void {
}
~/test$ sunder-compile test.sunder 
[test.sunder:7] error: redeclaration of `B` previously declared at [test.sunder:3]
    B;
    ^
```

**Comment by ashn-dot-dev** [comment]
Fixed in 3632325abb653b732227bf758eb7ea05f8b33c4e.
